imports.py

```diff
@@ -0,0 +1,203 @@
+"""Imports for Jupyter Notebook"""
+import os
+import json
+from typing import Dict, Union
+
+import httpx
+from IPython.display import display, HTML
+from httpx import Response
+
+
+BASE_URL = "http://20.119.216.32:8000/r4/"  # TIMS server
+REQUEST_HEADERS = {
+	"Content-Type": "application/fhir+json; charset=UTF-8",
+	"Accept-Charset": "utf-8",
+	"User-Agent": "HAPI-FHIR/5.0.0 (FHIR Client; FHIR 4.0.1/R4; apache)"}
+PROJECT_DIR = os.path.dirname(__file__)
+CACHE_DIR = os.path.join(PROJECT_DIR, 'cache')
+URI = str
+
+
+# Utils
+def encode_filename_str(x):
+	"""Convert such that can create friendly system filename"""
+	delim = '*'
+	return x.replace('/', f'{delim}slash')
+
+
+def decode_filename_str(x):
+	"""Convert from friendly system filename"""
+	delim = '*'
+	return x.replace(f'{delim}slash', '/')
+
+
+def url_and_params_to_filename(url: str, params: Dict) -> str:
+	"""Convert to filename"""
+	x = url
+	if params:
+		x += '___params_'
+		for k, v in params.items():
+			x += '__' + k + '_' + v
+	filename = encode_filename_str(x) + '.json'
+	return filename
+
+
+def cache_write(url: str, params: Dict, response: Dict):
+	"""Write to cache"""
+	filename = url_and_params_to_filename(url, params)
+	if not os.path.exists(CACHE_DIR):
+		os.makedirs(CACHE_DIR)
+	path = os.path.join(CACHE_DIR, filename)
+	with open(path, 'w') as f:
+		json.dump(response, f, indent=4)
+
+
+def cache_read(url: str, params: Dict) -> Union[Dict, None]:
+	"""Read from cache"""
+	filename = url_and_params_to_filename(url, params)
+	path = os.path.join(CACHE_DIR, filename)
+	if os.path.exists(path):
+		with open(path, 'r') as f:
+			data: Dict = json.load(f)
+		return data
+	return None
+
+
+# Proper funcs / classes
+def decode_response_status_not_ok(response: Response) -> bool:
+	"""Decode status"""
+	if response is None:
+		print("NO RESPONSE")
+		return True
+	else:
+		if 200 <= response.status_code < 300:
+			return False
+		elif 400 <= response.status_code < 500:
+			return True
+		elif response.status_code >= 500:
+			return True
+
+
+def decode_code_system_json(response: Dict):
```

Comment:
I changed a lot of your functions to accept dictionaries, sorry. This was just easier when reading straight from the cache. 
Example:
```python
def decode_code_system_json(response: Dict):
	"""Decode CodeSystem response"""
	# if not decode_response_status_not_ok(response):
	# 	response_dict = json.loads(response.text)
	if 'entry' in response:
```

Reply:
I didn't even notice. All you did was push the seam between functions over a bit. What I did notice is that you took out the test of the HTTP response.

Reply:
Yeah, I renamed `decode_status` to `decode_response_status_not_ok` and then commented out where it is used.

I personally don't mind if it shows an actual error message in the Jupyter output. I'm intending to share static HTML/PDF so they won't be running the notebook anyway. At least that's my assumption, so I could be wrong. I guess if I'm sharing screen and they see a stack trace I don't mind that either. I'd prefer to see why it is failing when I run the cell. But we can change it back to how you had it as well at some poitn if you prefer.